File: stack/valid-parentheses/valid-parentheses.ts
function isValid(s: string): boolean {
    const stack: string[] = [];

    for (const c of s) {
        if (c === '(' || c === '[' || c === '{') {
            stack.push(c);
        } else {
            const last = stack.pop();
            if (
                (c === ')' && last !== '(') ||
                (c === ']' && last !== '[') ||
                (c === '}' && last !== '{')
            ) {
                return false;
            }
        }
    }

    return stack.length === 0;
};

File: stack/valid-parentheses/valid-parentheses.cs
public class Solution
{
    public bool IsValid(string s)
    {
        if (string.IsNullOrEmpty(s) || s.Length == 1)
        {
            return false;
        }

        var bracketsDict = new Dictionary<char, char>()
        {
            {'(',')'},
            {'{','}'},
            {'[',']'},
        };

        var bracketsStack = new Stack<char>();
        for (int i = 0; i < s.Length; i++)
        {
            if (!bracketsStack.TryPeek(out var lastBracket))
            {
                bracketsStack.Push(s[i]);
                continue;
            }

            if (bracketsDict.ContainsKey(s[i]))
            {
                bracketsStack.Push(s[i]);
            }
            else if (bracketsDict[lastBracket] == s[i])
            {
                bracketsStack.Pop();
            }
            else
            {
                return false;
            }
        }

        if (bracketsStack.Count > 0)
        {
            return false;
        }

        return true;
    }
}